### ISSUE TITLE
[Flang] Missing error for MODULE prefix in abstract interface body

### ISSUE THREAD

**Issue by ejose02** [llvm/llvm-project#208200]
Flang fails to diagnose an invalid `MODULE` prefix on a function declared inside an ABSTRACT INTERFACE block.

Expected behavior: The code should be rejected at compile time. Per the Fortran standard, MODULE may appear only in the function or subroutine of a module subprogram, or of a nonabstract interface body declared in the scoping unit of a module or submodule.

Actual behavior: Flang compiles the program successfully with no diagnostic.

Reproducer:

```Fortran
Module mod_test
Abstract Interface
  Pure Integer Module Function f1(i)
    Integer, Intent(In) :: i
  End Function
End Interface
End Module

```

**Comment by eugeneepshteyn** [comment]
@ejose02 , thank you for the issue. I agree that flang should issue error in this case. Also, NAG, ifx, and nvfortran issue error in this case (but not gfortran). Do you plan on working on this, or should I leave it unassigned for others to pick up?

**Comment by ejose02** [comment]
> [@ejose02](https://github.com/ejose02) , thank you for the issue. I agree that flang should issue error in this case. Also, NAG, ifx, and nvfortran issue error in this case (but not gfortran). Do you plan on working on this, or should I leave it unassigned for others to pick up?

Thanks for the feedback. I'd like to work on this issue. It may take me some time to get a patch posted, but I plan to send an update once I have something ready for review.
